addons/calendar/models/calendar.py

```diff
@@ -715,7 +715,7 @@ def _get_display_time(self, start, stop, zduration, zallday):
                 2) if event all day ,return : AllDay, July-31-2013
         """
         timezone = self._context.get('tz') or self.env.user.partner_id.tz or 'UTC'
-        timezone = pycompat.to_native(timezone)  # make safe for str{p,f}time()
+        timezone = pycompat.to_text(timezone)  # make safe for str{p,f}time()
```

Comment:
Same as above I think, issue was strftime/strptime needs native strings, so we had to convert unicode back to str in P2.